### PR TITLE
setup content-id function per mime part #patch

### PR DESCRIPTION
**Pull request by davidblum**
Setup the `C_GO` binding for `content-id` on MIME part.

### CODE REVIEW

File: gmime/part.go [hunk 1]
```diff
@@ -103,6 +103,14 @@ func (p *Part) Headers() textproto.MIMEHeader {
 	return nil
 }
 
+// GetContentID returns the content ID (if there is one) per mime part
+// we expect all attachments to have a content-id
+func (p *Part) GetContentID() string {
+	cCID := C.g_mime_object_get_content_id(p.gmimePart)
```
Comment (Dharmik-Patel) [inline gmime/part.go:114]: you might need to something similar to filename function
`if p.gmimePart == nil {
		return ""
	}
	if !gobool(C.gmime_is_part(p.gmimePart)) {
		return ""
	}`

File: gmime/part.go [hunk 2]
```diff
@@ -103,6 +103,14 @@ func (p *Part) Headers() textproto.MIMEHeader {
...
+	defer C.g_free(C.gpointer(unsafe.Pointer(cCID)))
```
Comment (jimmyjames85) [inline gmime/part.go:115]: What happens guessing it returns empty string if there is not one? not too familiar with `C.g_free` stmt if we get null pointer back

File: gmime/part.go [hunk 3]
```diff
@@ -103,6 +103,14 @@ func (p *Part) Headers() textproto.MIMEHeader {
 	return nil
 }
 
+// GetContentID returns the content ID (if there is one) per mime part
+// we expect all attachments to have a content-id
```
Comment (Dharmik-Patel) [inline gmime/part.go]: there is also isattachment function if this is only expected on attachments https://github.com/sendgrid/go-gmime/pull/33/files#diff-e39577e7183b8f756ececf21707723e28ea58a5f06aa522703acd6e4552ab0dbR33

Comment (davidblum) [reply gmime/part.go]: hmmm, thats a good point.

This is how we implement it on our side.  I think its purpose should be generic and always check if it's requested on the `part`.

I'll update the description to reflect that.

File: gmime/part.go (hunk 1 above)
Comment (davidblum) [reply gmime/part.go:114]: Great call, and thank you for the code sample.

I'll get this updated.

**Review by jimmyjames85 — APPROVED** [review]
LGTM

File: gmime/part.go (hunk 2 above)
Comment (jimmyjames85) [reply gmime/part.go:115]: so looking at other spots in the code for null checks I fund this guy https://github.com/sendgrid/go-gmime/blob/539930aaa3d4ffe8abc1b882e93671c63a9ddf7c/gmime/part.go#L64
which calls this guy https://github.com/sendgrid/go-gmime/blob/539930aaa3d4ffe8abc1b882e93671c63a9ddf7c/gmime/gmime.c#L17
which can return null but the first guy doesn't do null check and defers the free stmt...

resolving comment

**Review by tbischel — APPROVED** [review]

**Pull request merged**